### DISCUSSION TITLE
GetLayout

### DISCUSSION THREAD

**Discussion by patrikha** [leftwm/leftwm#670] [Q&A] [answered]
Hi,

Would you consider adding a GetLayout command to leftwm-command? The story is that I would like to create a rofi applet for polybar that let me choose a layout from an on-screen list of options instead of cycle though the layout with the next and prev commands.
Just returning the layout name as string would be perfect.

cheers
Patrik

**Comment by VuiMuich** [comment, marked as answer]
The current layout is contained in the output of `leftwm state`.
You could either parse the `json` formatted string in your script or use `liquid` templating language to pre-parse the output. Here is a single line command to get the layout of `workspace 0`: `leftwm state -w0 -s "{{ workspace.layout }}"`

---
edit: typo in command

**Reply by patrikha** [reply]
OK that works fine, just for others ending up here, the command should be:
`leftwm-state -qs "{{ workspaces[0].layout }}"`

**Reply by VuiMuich** [reply]
Ooopsies, I had a typo in the command.
When the `-w 0` flag is used, the literal has to be `workspace` (singular)...

**Reply by patrikha** [reply]
Ended up with this command to get the layout of the currently active workspace:

`leftwm-state -qs "{% for workspace in workspaces %} {% for tag in workspace.tags %} {% if tag.mine and tag.visible and tag.focused %} {{ workspace.layout }} {% endif %} {% endfor %} {% endfor %}"`
